packages/experimental/Badge/src/CounterBadge/CounterBadge.tsx

```diff
@@ -28,12 +28,14 @@ export const CounterBadge = compose<CounterBadgeType>({
       const { count, icon, iconPosition = 'before', overflowCount, dot, ...mergedProps } = mergeProps(badge.props, final);
       const { showBadge } = badge.state;
       const displayCount = count && count > overflowCount ? `${overflowCount}+` : `${count}`;
+      const hasChildren = children[0] && children[0][0];
```

Comment:
>children[0] && children[0][0]; [](http://example.com/codeflow?start=26&length=30)

We could use React's Children API to check the children count:

`React.Children.count(children)`

Reply:
The only thing is, I'm not sure if it's good to check count, because even when there's no children passed, children prop will have value [[null]]